test-suite-genetation.md

```diff
@@ -0,0 +1,131 @@
+A drawback of such approach is that there can be conflicting testing goals and it might be impossible to cover all of them with a single test sequence, regardless of its length.  
+Another method proposed optimizes entire test suites with a search algorithm with respect to mutation analysis. 
+However, the strong limitation is having to manually choose and fix the length of the test cases, which does not evolve during the search.  
+To deal with these drawbacks, the paper therefore approaches a new method with the intent to satisfy the chosen coverage criterion with the smallest possible test suite, which makes use of a search algorithm applied on a population of test suites. 
+## Test Suite Optimization
+To execute whole test suite generation, the paper describes the use of a Genetic Algorithm (GA) as search algorithm, on a population of test suites. It also focuses on *branch coverage* as test criterion. 
+
+### Genetic Algorithm
+This algorithm starts with a random population of test suites, and evolves this population until a solution is found that fulfills the coverage criterion, or the set resources such as time or number of fitness evaluations have been used up. 
+For each generation, the new population is initialised with the best individuals of the last generation. 
+The new generation is then filled up with individuals produced by rank selection, crossover, and mutation. 
+Either the offspring or the parents are added to the new generation, depending on fitness and length constraints.   
+
+### Representation
+To apply this search algorithm, the problem needs to be defined, and a solution for that problem needs to be represented accurately. 
+The paper declares the solution to be a *test suite*, which is represented as a set of test cases. 
+A test case is represented by a sequence of statements, and the length of the test suite is defined as the sum of the lengths of its test cases. 
+Test cases contain statements, and each statement has a value and a type. This paper identifies five types of statements:
+- **Primitive** statements, which represent Boolean, String, and enumeration variables, but also arrays of any type.
+- **Constructor** statements, which generate new instances of any given class.
+- **Field** statements, which access public member variables of objects. 
+- **Method** statements, which invoke methods on objects or call static methods.
+- **Assignment** statements, which assign values to array indices or to public member variables of objects, but do not define new values.
+
+The *test cluster* for a given system under test defines the set of available classes, their public constructors, methods, and fields. 
+This chosen representation has a variable size for the test suite for one good reason: for a new software to test, the optimal number of test cases and their respective optimal length are not known, and need to be searched for. 
```

Comment:
statements, and each  - no comma
And better to put a stop here ...to test. The optimal